### DISCUSSION TITLE
EGSnrc Test

### DISCUSSION THREAD

**Discussion by CuiHenry** [nrc-cnrc/EGSnrc#1120] [Show and tell]
Hi，guys：
I defined a simple simulation using EGSnrc to study photon transport and dose distribution in Tungsten and aqueous media.
But after i run it, the results don't feel right.
![1713161371462](https://github.com/nrc-cnrc/EGSnrc/assets/167058944/eafe13e7-305e-4eaf-a265-a365ca578ae5)
Is the deposition energy of individual particles or the total deposition energy of the medium shown?
The attachment is my code.
[slab.zip](https://github.com/nrc-cnrc/EGSnrc/files/14974788/slab.zip)
Thanks!!!!

**Comment by rtownson** [comment]
This is due to the normalization of the collimated source, which is different from many of the other sources in EGSnrc. It is explored in the getting-started manual, but here is a better description:

https://nrc-cnrc.slides.com/ftessier/egsnrc-collimated-source/fullscreen

The TLDR is just to set `distance` in the collimated source to be `1/sqrt(4*Pi)` (calculate the actual value and use that). This will normalize the results to be "per particle" as you expect. It does assume that you are modelling an isotropic source and just collimating it.

**Reply by CuiHenry** [reply]
Thanks!
And i have an other problem.
I emit 1000 photons of 9Mev, why is the energy deposited on the tungsten plate 15.563Mev, which is greater than my defined energy of 9Mev?

**Reply by rtownson** [reply]
Same answer - after you set `distance = 0.28209479177` you will see this drop. In the input file you sent me, I got `1.252 MeV` after the change.

**Reply by CuiHenry** [reply]
Thanks！I got it.

**Reply by CuiHenry** [reply]
Hi, Rtownson:
If i use：Typical Energy: 9MEV
Dose Rate: ~1K-3K R/min depending on application.
SDD: 170.4” environment to test.
Am I writing this right?
Code:
[slab.zip](https://github.com/nrc-cnrc/EGSnrc/files/15020106/slab.zip)

**Reply by rtownson** [reply]
Your setup looks about right, just keep in mind that these inputs do not do anything, so you should remove all of this:
```
:particle fluence:
   particle = photon
   fluence = 1000 3000
   energy = 9
   
   # 修改距离为 1/sqrt(4*pi)
:distance normalization:
```

You can't set dose rate in the simulation; there are no time dependent effects modelled. Just apply a scaling factor to the results.

**Reply by CuiHenry** [reply]
If use beamnrc_gui.tcl, can i set these Dose Rate:~1K-3K R/min?
I didn't find that how to set it.

**Reply by CuiHenry** [reply]
If i use beamnrc_gui or dosxyznrc_gui, how can i realization of the above simulation environment.
Many thanks!

**Reply by CuiHenry** [reply]
[slab.zip](https://github.com/nrc-cnrc/EGSnrc/files/15033588/slab.zip)
Hi, I update it again, can you help to see that the code is right?
And i want to know that what is the energy residue after passing through air, tungsten plates, and water respectively.
I write some code but it seem not work.

**Reply by rtownson** [reply]
There is no parameter for dose rate anywhere in EGSnrc - it is not necessary because it has no impact on a simulation that reports results per initial particle. You can just apply a normalization after the simulation.

It's not possible to guess at parameters, so none of these extra input blocks that you added will do anything:
```
# 添加能量残余对象以查看能量分布
:start ausgab object:
name = energy_residual
library = egsSpectrumAnalysis
energy residual regions = 1 2 # 指定要计算的介质（例如，1表示Tungsten，2表示水）
:stop ausgab object:

# 添加能量积分对象以查看能量分布
:start ausgab object:
name = energy_integration
library = egsPhspFluence
energy deposition regions = 1 2 # 指定要计算的介质（例如，1表示Tungsten，2表示水）
:stop ausgab object:
```
Please use the [documentation](https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/index.html) to see what is available.

Consider using [phase-space scoring](https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/classEGS__PhspScoring.html), or [fluence scoring](https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/classEGS__PlanarFluence.html), carefully following the documentation.

Edit: Also note that you can only have one `start ausgab object definition`. Same goes for all of the `definition` input blocks. It will just use the first and ignore any others.

**Reply by CuiHenry** [reply]
Ok!
I noticed the file tutor2.mortran, which describes what I want to run, but how do I run it? What do I type?
![1713781149980](https://github.com/nrc-cnrc/EGSnrc/assets/167058944/00a01d78-4cdc-4380-89a3-9a8d5972068a)

[tutor2.zip](https://github.com/nrc-cnrc/EGSnrc/files/15061394/tutor2.zip)

Many thanks!

**Reply by rtownson** [reply]
After compiling, run with `tutor2 -p tutor_data`. Note there is also `tutor2pp`, which is more flexible because you can change the geometry and source without editing the code.

**Reply by CuiHenry** [reply]
Maybe i have some trouble. I can't run it.
![image](https://github.com/nrc-cnrc/EGSnrc/assets/167058944/db14385d-9fe4-4da2-8464-7c2acc1ff8e1)
Can you help me why i can't run it?
Thanks!

**Reply by rtownson** [reply]
Looks like you haven't compiled it. To compile on the command line, in the directory type `mingw32-make`.

**Reply by CuiHenry** [reply]
I'm so sad because i can't compiled it successful.
In fact i try any ways to compile it but also filed.
[tutor2.zip](https://github.com/nrc-cnrc/EGSnrc/files/15229087/tutor2.zip)
The code is wrong?

**Reply by rtownson** [reply]
I don't recommend using tutor2, the same functionality is available in tutor2pp but then you can use an egs++ style input file. I think in MORTRAN it will be much more challenging if you want to change anything.

To get the code to compile, remove the line `DATA  MEDARR /$S'W',22*' '/;`, and change the line  `DO I=1,24[MEDIA(I,1)='W';]`. Remove your comments at the top that start with `#`, they need to use quotes instead. The line you changed `/UIN,VIN/=0.0;WIN=6.0;` I think does not do what you expect - these are direction cosines, not positions, so they should add to `1`.

**Reply by CuiHenry** [reply]
Thanks a lot!
I can use tutor2pp to realize this result? (In fact I didn't find sample code to do this in tutor2pp)
![image](https://github.com/nrc-cnrc/EGSnrc/assets/167058944/d6a037be-0e0c-4716-b8fa-0320ea436e5f)
I want to analog energy accounted.
Can you send me a sample code for my reference?

**Reply by CuiHenry** [reply]
ohh! I find it.
Sorry i'm so carelessness.

**Reply by CuiHenry** [reply]
![image](https://github.com/nrc-cnrc/EGSnrc/assets/167058944/23cae95c-223f-4be1-99a8-1cfe3b3830a5)
I run it but why it don't show me energy accounted like the last image?
[test1.zip](https://github.com/nrc-cnrc/EGSnrc/files/15243238/test1.zip)

**Reply by rtownson** [reply]
My apologies, it looks like tutor2pp has been broken for a while. It has a segmentation fault. I think you're the first person to use it in a long time! I suppose either use tutor2, or tutor7pp. If you're looking for those deposited, transmitted and reflected quantities, tutor7pp does this as well.

**Reply by CuiHenry** [reply]
[tutor7pp.zip](https://github.com/nrc-cnrc/EGSnrc/files/15271356/tutor7pp.zip)
OK! 
I use tutor7pp, but i can't change the media = tungsten.
Unable to recognize new media name.
And with use egs_box, i can't make sure the right setting with 1 cm thick tungsten plate, because box size = 1 1 435.
It means 435mm is tungsten thickness?
 
Can you help to run it?

**Reply by CuiHenry** [reply]
I want to set the light source position at 435 from the plate 435.
How can i  do ?

**Reply by rtownson** [reply]
Have you done the getting started manual? It will help you learn how to set materials and design geometries. If you are using `-p tutor_data`, it doesn't contain tungsten. Instead, learn how to use pegsless materials. E.g.

```
:start media definition:

    ### cutoff energies
    ae  = 0.521            # lowest  energy for electrons (kinetic+0.511)
    ap  = 0.01              # lowest  energy for photons   (kinetic)
    ue  = 30.511            # maximum energy for electrons (kinetic+0.511)
    up  = 30                # maximum energy for photons   (kinetic)

    :start air:
        density correction file    = air_dry_nearsealevel
    :stop air:

    :start tungsten:
        density correction file     = tungsten
    :stop tungsten:
:stop media definition:
```

Instead of egs_box you could use planes like you were in tutor2pp, and the same source design.

**Reply by CuiHenry** [reply]
I changed but it show me EGS_Application::initSimulation(): scoring initialization failed with status 4804176.
Use:
################################
### AUSGAB OBJECTS
################################
:start ausgab object definition:

    :start ausgab object:

        library = egs_track_scoring
        score photons   = yes
        score electrons = yes
        score positrons = yes

    :stop ausgab object:

:stop ausgab object definition:

Can you tell me why?

[testapp.zip](https://github.com/nrc-cnrc/EGSnrc/files/15281291/testapp.zip)

**Reply by rtownson** [reply]
For tutor7pp you need to at least have a scoring options input block, even if you don't use it. There are some inputs to allow you to score spectra. Just add these lines:
```
:start scoring options:
:stop scoring options:
```
